artiq/coredevice/comm_kernel.py

```diff
@@ -476,11 +476,27 @@ def check(cond, expected):
             if tag_element == "b":
                 self._write(bytes(value))
             elif tag_element == "i":
-                self._write(struct.pack(self.endian + "%sl" %
-                                        len(value), *value))
+                try:
+                    data = struct.pack(self.endian + "%sl" % len(value), *value)
+                except struct.error:
+                    raise RPCReturnValueError(
```

Comment:
What problem is that supposed to solve?

Reply:
This used to be checked one by one before
but it comes at a great performance penalty cost (see the benchmark above)
and we found the root cause in #1801, so we should translate `struct.error` instead
so as to keeping the same behavior of what the check functions did in serializing the primitive types above, aka throwing an RPCReturnValueError error

Reply:
I don't understand

Reply:
maybe I should move this hunk to #1801 as well?

Reply:
I don't see why it should be there at all.

Reply:
https://github.com/m-labs/artiq/blob/d7dd75e833fc99ad75e10220a5ec8f03d64ee5bd/artiq/coredevice/comm_kernel.py#L438-L447
I was referring to this part of the code, shouldn't we check if all elements in TList[int]/TList[int64] are within bound as well? Otherwise it will just throw `struct.error` on the array roundtrip test, which is weird.

Reply:
OK, I see. Should be a separate PR anyway.